force-app/main/default/classes/Addresses.cls

```diff
@@ -485,69 +569,69 @@ public inherited sharing class Addresses extends fflib_SObjects {
+        NPSP_Address npspAddress = new NPSP_Address(address, oldAddress);
 
-    public Map<Id, Address__c> nonDefaultsWithChanges() {
-        Map<Id, Address__c> nonDefaultsWithChanges = new Map<Id, Address__c>();
-        Integer i = 0;
-        for (Address__c addr : (List<Address__c>) getRecords()) {
-            NPSP_Address npspAddress = new NPSP_Address(addr);
-            Address__c addrOld = oldVersionOf(addr);
-            NPSP_Address npspAddressOld = new NPSP_Address(addrOld);
-            if ((!npspAddress.isEqualToIncludingAddressType(npspAddressOld, TRUE) ||
-                isUndeliverableStatusChanged(addr, addrOld))) {
+        if (oldAddress == null) {
+            return npspAddress.isDefault() ||
+                    npspAddress.isSeasonalAddr() ||
+                    (npspAddress.isUndeliverable() && npspAddress.isDefault());
+        }
+
+        if (npspAddress.isNewDefault() ||
+            isSeasonalChanged(address, oldAddress) ||
+            (npspAddress.isUndeliverableStatusChanged()) && npspAddress.isDefault()) {
+
+            return true;
+        } else if (!npspAddress.isEqualToIncludingAddressType(npspAddress.oldVersion(), TRUE)) {
+            if (npspAddress.isDefault()) {
 
-                nonDefaultsWithChanges.put(idOf(addr), addr);
+                return true;
             }
-            i++;
         }
-        return nonDefaultsWithChanges;
+        return false;
     }
 
-    public Map<Id, Address__c> addressesByAccountId() {
-        Map<Id, Address__c> mapAddressesByAcctId = new Map<Id, Address__c>(); // map of HH Accounts, and their default Address
+    public Map<Id, Address__c> addressChangesToSyncToContactOverrides() {
+        Map<Id, Address__c> addressesById = new Map<Id, Address__c>();
         for (Address__c address : (List<Address__c>) getRecords()) {
-            if (hasHouseholdAccount(address)) {
-                mapAddressesByAcctId.put(householdAccountIdFor(address), address);
+            NPSP_Address npspAddress = new NPSP_Address(address, oldVersionOf(address));
+
+            if (shouldSyncAddressChangesToContactOverrides(npspAddress)) {
+                addressesById.put(npspAddress.getId(), npspAddress.getRecord());
             }
         }
-        return mapAddressesByAcctId;
+        return addressesById;
     }
 
-    public Map<Id, Address__c> defaultAndSeasonalAddressesByHouseholdAccountId() {
-        Map<Id, Address__c> defaultAndSeasonalAddressesByAcctId = new Map<Id, Address__c>();
-        for (Address__c addr : (List<Address__c>) getRecords()) {
-            NPSP_Address npspAddress = new NPSP_Address(addr);
-            if (isDefault(addr) || npspAddress.isSeasonalAddr()) {
-                defaultAndSeasonalAddressesByAcctId.put(householdAccountIdFor(addr), addr);
+
+    private Boolean shouldSyncAddressChangesToContactOverrides(NPSP_Address npspAddress) {
```

Comment:
Can this instead be an instance method on NPSP_Address? I think `hasChangesToSync()` would be an appropriate name for it over there.